File: Polygon.java
public class Polygon extends Shape  {
    protected int[] sideLengths;

    public Polygon() {
        super();
        this.shape = "polygon";
        this.sideLengths = null;
    }
    
    public int computePerimeter() {
        if (this.sideLengths == null) {
            return 0;
        }
        int perimeter = 0;
        for (int length : this.sideLengths) {
            perimeter += length;
        }
        return perimeter;
    }

    public int getNumberOfEdges() {
        if (this.sideLengths == null) {
            return 0;
        }
        return this.sideLengths.length;
    }
}

File: polygon.py
from shape import Shape

class Polygon(Shape):
    def __init__(self):
        super().__init__()  # Call the parent class's initializer
        self.shape = 'polygon'
        self.side_lengths = None

    def compute_perimeter(self):
        if self.side_lengths is None:
            return 0
        return sum(self.side_lengths)

    def get_number_of_edges(self):
         if self.side_lengths is None:
            return 0
         return len(self.side_lengths)